# Bioinformatics

<!-- START doctoc generated TOC please keep comment here to allow auto update -->
<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
**Table of Contents**

- [Bioinformatician](#bioinformatician)
- [Bioinformatics](#bioinformatics-1)
- [生物信息杂谈](#%E7%94%9F%E7%89%A9%E4%BF%A1%E6%81%AF%E6%9D%82%E8%B0%88)
- [Talks](#talks)
- [Online courses](#online-courses)
- [Workshop](#workshop)
- [Comprehensive packages](#comprehensive-packages)
- [General file formats](#general-file-formats)
- [bam/sam/tabix/bgzf](#bamsamtabixbgzf)
- [Fasta/q](#fastaq)
- [GFF/BED/VCF](#gffbedvcf)
- [Others formats](#others-formats)
- [Database API](#database-api)
- [data structure](#data-structure)
- [Models](#models)
- [Scripts](#scripts)
- [Visualization](#visualization)
    - [Circos Related](#circos-related)
    - [Others](#others)
- [Kmer](#kmer)
- [Phylogenetic tree](#phylogenetic-tree)
- [Taxonomy](#taxonomy)
- [Assembly](#assembly)
- [Alignment](#alignment)
- [Multiple Alignment](#multiple-alignment)
- [Mapping](#mapping)
- [Bacterial comparative genomics](#bacterial-comparative-genomics)
- [Metagenomics](#metagenomics)
- [16S](#16s)
- [Classifier | removing human reads](#classifier--removing-human-reads)
- [Virome](#virome)
- [Chip-seq](#chip-seq)
- [Plastform](#plastform)
- [PCR](#pcr)
- [HPC](#hpc)
- [Transcriptome](#transcriptome)
- [Variant Calling](#variant-calling)

<!-- END doctoc generated TOC please keep comment here to allow auto update -->


## Bioinformatician

-  [What is a bioinformatician](http://blog.fejes.ca/?p=2418)
-  [Benjamin Franklin Award for Open Access in the Life Sciences](http://www.bioinformatics.org/franklin/)
-  [My Formula as a Bioinformatician](http://computationalproteomic.blogspot.co.uk/2014/01/my-formula-as-bioinformatician.html)
-  [So you want to be a computational biologist?](http://www.nature.com/nbt/journal/v31/n11/abs/nbt.2740.html) ☆
-  [Bioinformatics is not something you are taught, it’s a way of life](http://biomickwatson.wordpress.com/2013/08/06/bioinformatics-is-not-something-you-are-taught-its-a-way-of-life/)
-  [A guide for the lonely bioinformatician](http://www.opiniomics.org/a-guide-for-the-lonely-bioinformatician/)
-  [Bioinformatics Curriculum Guidelines: Toward a Definition of Core Competencies](http://www.ploscompbiol.org/article/info:doi%2F10.1371%2Fjournal.pcbi.1003496)
-  [Top N Reasons To Do A Ph.D. or Post-Doc in Bioinformatics/Computational Biology](http://caseybergman.wordpress.com/2012/07/31/top-n-reasons-to-do-a-ph-d-or-post-doc-in-bioinformaticscomputational-biology/)
-  [101 Questions: a new series of interviews with notable bioinformaticians](http://www.acgt.me/?tag=101+questions)
-  [生物信息学家级别](http://blog.sciencenet.cn/blog-404304-834869.html) [Levels of Bioinformatics Research](http://www.longwoodgenomics.org/2014/10/11/levels-of-bioinformatics-research/)
-  [It’s time to reboot bioinformatics education](http://toddharris.net/blog/2015/03/23/its-time-to-reboot-bioinformatics-education/)
-  [An Explosion Of Bioinformatics Careers](http://sciencecareers.sciencemag.org/career_magazine/previous_issues/articles/2014_06_13/science.opms.r1400143)
-  [**Is going back to the wet lab worth it?**](https://www.biostars.org/p/228950/#231887)
- [5 things I wish I knew when I started getting into bioinformatics](https://angus.readthedocs.io/en/2019/_static/Lee-5-things.pdf)

Social media

- [Staying Current in Bioinformatics & Genomics: 2017 Edition](http://www.gettinggeneticsdone.com/2017/02/staying-current-in-bioinformatics-genomics-2017.html)
- [Interesting bioinformatics blogs (2017 edition)](https://www.biostars.org/p/243961/)

Programming skills

- [Linux Command Line for Bioinformatics](https://telatin.github.io/microbiome-bioinformatics/Bash-course/)
- [An Introduction to Programming for Bioscientists: A Python-Based Primer](http://journals.plos.org/ploscompbiol/article/asset?id=10.1371%2Fjournal.pcbi.1004867.PDF)
-  [You can code, too!](http://blog.olgabotvinnik.com/blog/2013/11/22/2013-11-22-you-can-code-too/)

## Bioinformatics

-  [The Phylogeny of Everything, the Origin of Eukaryotes, and the Rules of Taxonomy: Death to Archaea, Bacteria, and Eucarya! Long live Archaebacteria, Eubacteria, Eukaryota, and Prokaryota!](http://judgestarling.tumblr.com/post/79135526488/the-phylogeny-of-everything-the-origin-of-eukaryotes)
-  [Crossroads (iii) – a New Direction for Bioinformatics with Twelve Fundamental Problems](http://www.homolog.us/blogs/blog/2014/10/14/crossroads-iii-a-new-direction-for-bioinformatics-with-twelve-fundamental-problems/)
-  [Ten Simple Rules for Effective Computational Research](http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003506)（[高效计算科学研究的十条简单规则](/ten-simple-rules-for-effective-computational-research/)）
-  [Ten Simple Rules for Reproducible Computational Research](http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003285)
-  [Ten Simple Rules for the Care and Feeding of Scientific Data](http://www.ploscompbiol.org/article/info%3Adoi%2F10.1371%2Fjournal.pcbi.1003542)
-  [An Introduction To Applied Bioinformatics](http://caporasolab.us/An-Introduction-To-Applied-Bioinformatics/)
-  [Freedom in bioinformatics](http://journal.frontiersin.org/Journal/10.3389/fgene.2014.00259/full)
-  [二代测序数据辨（之一）：Clean Data](http://garification.blog.163.com/blog/static/1741330252014624114724457/)
-  [病原微生物高通量测序数据分析笔记](http://indexofire.gitbooks.io/notebook_of_analyzing_pathogen_ngs_data/)
-  [What to do with lots of (sequencing) data](http://ivory.idyll.org/blog//2015-what-to-do-with-sequencing-data.html)
-  [The myths of bioinformatics software](https://liorpachter.wordpress.com/2015/07/10/the-myths-of-bioinformatics-software)
-  [**Good Habit for Bioinformatics Analyst or Scientist**](https://www.biostars.org/p/190366/)
-  [**What Are The Most Common Stupid Mistakes In Bioinformatics?**](https://www.biostars.org/p/7126/#191667)
-  [Myths about Bioinformatics](https://bioinformaticsreview.com/20200302/myths-about-bioinformatics/)

## 生物信息杂谈

-  [《学生物的，不会编程，也可以报考生物信息学的研究生》](http://blog.sciencenet.cn/home.php?mod=space&uid=61772&do=blog&id=590128)by [牛登科](http://blog.sciencenet.cn/home.php?mod=space&uid=61772)。（学生物的，不会编程，也能学生物信息学技术）
-  《[高通量测序能替代PCR吗？](http://blog.sina.com.cn/s/blog_52cb75b90102ekvq.html)》 by [韩建](http://blog.sina.com.cn/jianhan1)
-  [《生物信息学数据分析与皇帝的新装》](http://blog.sciencenet.cn/blog-604918-582915.html)
-  [个性化医疗会带来更昂贵的药物？](http://www.biodiscover.com/news/industry/109583.html)
-  [高通量测序公司靠什么赚钱？](http://blog.sina.com.cn/s/blog_52cb75b90102uw21.html)
-  [生物不退学指南：教你如何靠生物养家糊口](http://mp.weixin.qq.com/s?__biz=MzA3MjQzODAzNA==&mid=200430076&idx=1&sn=35635d8a4b1c13be069ee557f7466983&scene=2&from=timeline&isappinstalled=0#rd) （想进入生物学领域的请看）

## Talks

- [How NGS is transforming medicine](http://genomespot.blogspot.be/2017/02/how-ngs-is-transforming-medicine.html)


## Online courses

-  [https://liulab-dfci.github.io/bioinfo-combio/](https://liulab-dfci.github.io/bioinfo-combio/)
-  [Rosalind](http://rosalind.info/) (Rosalind is a platform for learning bioinformatics through problem solving)
-  [**Teaching Materials of Langmead-lab**](http://www.langmead-lab.org/teaching-materials/)

## Workshop

- [Next-Gen Sequence Analysis Workshop](https://angus.readthedocs.io/en/2019/)

## Book

- [A Primer for Computational Biology](http://library.open.oregonstate.edu/computationalbiology/)
- [Human Genome Variation Lab](https://mccoy-lab.github.io/hgv_modules/index.html), teaching materials from our undergrad computational course on human genetic variation

## Comprehensive packages

- [python] [Biopython](http://biopython.org/)
- [golang] [Biogo](https://github.com/biogo)
- [golang] [bio](https://github.com/shenwei356/bio) - A simple but high-performance bioinformatics package in Go

## Sequencing

### About reads duplicates

> I recommend optical duplicate removal for all HiSeq platforms, for any kind of project in which you expect high library complexity (such as WGS). By optical duplicate, I mean removal of duplicates with very close coordinates on the flow cell

- [Duplicates on Illumina](https://www.biostars.org/p/229842/)
- [Remove duplicates from reads: best practices?](https://www.biostars.org/p/230822/)
- [`bbmap clumpify` can remove PCR and optical duplicates](https://www.biostars.org/p/225338/#230178)
- [Deduplication Improves Cost-Efficiency and Yields of De Novo Assembly and Binning of Shotgun Metagenomes in Microbiome Research](https://journals.asm.org/doi/10.1128/spectrum.04282-22)

## General file formats

- [zindex](https://github.com/mattgodbolt/zindex) - Create an index on a compressed text file
- [tabix](https://github.com/samtools/tabix)  - table file index
- [wormtable](https://github.com/wormtable/wormtable) - Write-once-read-many table for large datasets.

## bam/sam/tabix/bgzf

- [python] [hts-python](https://github.com/brentp/hts-python) - pythonic wrapper for libhts
- [python] [htseq](https://github.com/simon-anders/htseq) - HTSeq is a Python library to facilitate processing and analysis of data from high-throughput sequencing (HTS) experiments. http://www-huber.embl.de/users/anders/HTSeq/
- [golang] [biogo/hts](https://github.com/biogo/hts)
- [bamtools](https://github.com/pezmaster31/bamtools) - C++ API & command-line toolkit for working with BAM data
- [samblaster](https://github.com/GregoryFaust/samblaster) -  a tool to mark duplicates and extract discordant and split reads from sam files.
- [python] [pysamstats](https://github.com/alimanfoo/pysamstats) - A fast Python and command-line utility for extracting simple statistics against genome positions based on sequence alignments from a SAM or BAM file.
- [python] [pysam](https://github.com/pysam-developers/pysam) - a python module for reading and manipulating Samfiles. It's a lightweight wrapper of the samtools C-API. Pysam also includes an interface for tabix. Another sam parser: [simplesam](https://github.com/mdshw5/simplesam)
- [grabix](https://github.com/arq5x/grabix) - a wee tool for random access into BGZF files
- [golang]  [bix](https://github.com/brentp/bix) - tabix file access with golang using biogo machinery
- [mergesam](https://github.com/DarwinAwardWinner/mergesam) - Automate common sam & bam conversions
- [SAMstat](http://samstat.sourceforge.net/) - Displaying sequence statistics for next generation sequencing

## Fasta/q

- [seqtk](https://github.com/lh3/seqtk) - Toolkit for processing sequences in FASTA/Q formats
- [seqkit](http://github.com/shenwei356/seqkit) - A cross-platform and efficient toolkit for FASTA/Q file manipulation http://bioinf.shenwei.me/seqkit
- [python] [pyfaidx](https://github.com/mdshw5/pyfaidx) - pyfaidx: efficient pythonic random access to fasta subsequences
- [golang] [bio](https://github.com/shenwei356/bio) - A lightweight and high-performance bioinformatics package in Go

FASTA index

- [golang] [faidx](https://github.com/brentp/faidx)
- [golang] [bio/seqio/fai](https://github.com/shenwei356/bio/tree/master/seqio/fai)

## GFF/BED/VCF

- [bedtools2](https://github.com/arq5x/bedtools2) - A powerful toolset for genome arithmetic.
- [BEDOPS](https://bedops.readthedocs.org/en/latest/index.html) - the fast, highly scalable and easily-parallelizable genome analysis toolkit
- [gffcompare](https://github.com/gpertea/gffcompare) - classify, merge, tracking and annotation of GFF files by comparing to a reference annotation GFF
- [gffread](https://github.com/gpertea/gffread) - GFF/GTF utility providing format conversions, region filtering, FASTA sequence extraction and more

- [python] [gffutils](https://github.com/daler/gffutils) - GFF and GTF file manipulation and interconversion
- [python] [pybedtools](https://github.com/daler/pybedtools) - Python wrapper for Aaron Quinlan's BEDTools
- [golang] [irelate](https://github.com/brentp/irelate) - Streaming relation (overlap, distance, KNN) of (any number of) sorted genomic interval sets. #golang
- [golang] [vcfgo](https://github.com/brentp/vcfgo) - a golang library to read, write and manipulate files in the variant call format.
- [vcflib](https://github.com/vcflib/vcflib) - a simple C++ library for parsing and manipulating VCF files, + many command-line utilities 


## Others formats

- [blast_table2xml](https://github.com/jameslz/blast_table2xml) - Convert blast m6 format to xml for blast2go
- [*seqmagick*](http://seqmagick.readthedocs.org/en/latest/) - file format conversion in Biopython in a convenient way

## Database API

- [pyensembl](https://github.com/hammerlab/pyensembl) - Python interface to ensembl reference genome metadata (exons, transcripts, etc...)

## data structure

- [kvector](https://github.com/olgabot/kvector) - kvector is a small utility for converting motifs to kmer vectors to compare motifs of different lengths

## Models

- [pomegranate](https://github.com/jmschrei/pomegranate) - Graphical models for Python, implemented in Cython for speed.

## Scripts

- [oneliners](https://github.com/stephenturner/oneliners) - Useful bash one-liners for bioinformatics.
- [cgat](https://github.com/CGATOxford/cgat) - CGAT - Computational Genomics Analysis Tools
- [bcbb](https://github.com/chapmanb/bcbb) - Incubator for useful bioinformatics code, primarily in Python and R [http://bcbio.wordpress.com](http://bcbio.wordpress.com/)
- [jcvi](https://github.com/tanghaibao/jcvi) - Python utility libraries on genome assembly, annotation and comparative genomics
- [picobio](https://github.com/peterjc/picobio) - Miscellaneous Bioinformatics scripts etc mostly in Python
- [pydna](https://github.com/BjornFJohansson/pydna) - Classes and code for representing double stranded DNA and functions for simulating homologous recombination and Gibson assembly.
- [BioUtils](https://github.com/likit/BioUtils) - Python scripts for miscellaneous bioinformatics stuff.
- [sesbio](https://github.com/sestaton/sesbio) - Bioinformatics scripts for genome analysis
- [ngsutils](https://github.com/ngsutils/ngsutils) - Tools for next-generation sequencing analysis [http://ngsutils.org](http://ngsutils.org/)
- [ngsTools](https://github.com/mfumagalli/ngsTools) - Programs to analyse NGS data for population genetics purposes

## Visualization

- [Circleator](https://github.com/jonathancrabtree/Circleator) - Flexible circular visualization of genome-associated data with BioPerl and SVG.
- [ComplexHeatmap](https://github.com/jokergoo/ComplexHeatmap) - make complex heatmaps as well as self define annotation graphics
- [dalliance](https://github.com/dasmoth/dalliance) - Interactive web-based genome browser. http://www.biodalliance.org/
- [Question: Which program, tool, or strategy do you use to visualize genomic rearrangements?](https://www.biostars.org/p/158039/) 
- [DNAplotlib](https://github.com/VoigtLab/dnaplotlib) - DNA plotting library for Python

### Circos Related

- [Circos](http://circos.ca/): Perl package for circular plots, which are well suited for genomic rearrangements.
- [J-Circos](http://www.australianprostatecentre.org/research/software/jcircos) is a java application for doing interactive work with circos plots.
- [ClicO FS](http://bioinformatics.oxfordjournals.org/content/early/2015/08/18/bioinformatics.btv433.long): an interactive web-based service of Circos.
- [rCircos](https://cran.r-project.org/web/packages/RCircos/index.html): R package for circular plots. [last update: 2013]
- [OmicCircos](http://www.bioconductor.org/packages/release/bioc/html/OmicCircos.html): R package for circular plots for omics data.[last update: 2015-04]

### Gene Annotation

- [Gviz](https://bioconductor.org/packages/release/bioc/html/Gviz.html) - Plotting data and annotation information along genomic coordinates
- [pyGenomeTracks](https://github.com/deeptools/pyGenomeTracks) - python module to plot beautiful and highly customizable genome browser tracks
- [karyoploteR](https://github.com/bernatgel/karyoploteR) - karyoploteR - An R/Bioconductor package to plot arbitrary data along the genome
- [gggenes](https://github.com/wilkox/gggenes) - Draw gene arrow maps in ggplot2
- [DnaFeaturesViewer](https://github.com/Edinburgh-Genome-Foundry/DnaFeaturesViewer) - Python library to plot DNA sequence features (e.g. from Genbank files)

### Others

- [ggbio](http://bioconductor.org/packages/release/bioc/html/ggbio.html): R package for visualizing biological data. Has a circular view similar to the previous packages.
- [D3 chord diagrams](https://github.com/mbostock/d3/wiki/Chord-Layout) (javascript) can be used to visualize genomic rearrangements. [See this plot of migration flows](http://www.global-migration.info/) as a similar example.
- [Genomatix Transcriptome Viewer: Gene Fusion analyses](http://www.genomatix.de/online_help/help_eldorado/transcriptome_help.html#gene_fusion_analyses)
- [iFUSE: integrated fusion gene explorer](http://www.ncbi.nlm.nih.gov/pubmed/23661695)
- [FusionAnalyser: a new graphical, event-driven tool for fusion rearrangements discovery](http://www.ncbi.nlm.nih.gov/pubmed/22570408)
- [SOAPFuse includes the option to generate figures](http://soap.genomics.org.cn/soapfuse.html#output2)
- [Gremlin](http://compbio.cs.brown.edu/projects/gremlin/)
- [Seqeyes](http://www.biovis.net/2011/materials/abstracts/BioVispaper107.pdf): A flash tool for visualizing structural variations.
- [SVVIZ](http://svviz.github.io/svviz/) - A READ VISUALIZER TO VALIDATE STRUCTURAL VARIANTS
- [samplot](https://github.com/ryanlayer/samplot) - Plot structural variant signals from many BAMs and CRAMs
- [Understanding UMAP](https://pair-code.github.io/understanding-umap/)

## Kmer

- [khmer](https://github.com/dib-lab/khmer) - In-memory nucleotide sequence k-mer counting, filtering, graph traversal and more http://khmer.readthedocs.org/
- [Jellyfish](https://github.com/gmarcais/Jellyfish)
 


## Phylogenetic tree

- [R] [ggtree](http://www.bioconductor.org/packages/3.1/bioc/vignettes/ggtree/inst/doc/ggtree.html) - a phylogenetic tree viewer for different types of tree annotations
- [python] [ETE tools](http://etetoolkit.org/)
- [evolview](http://evolgenius.info/evolview/)

## Taxonomy

- [NCBI_taxonomy_tree](https://github.com/frallain/NCBI_taxonomy_tree) - NCBI taxonomy tree in-memory mapping
- [taxiphy](https://github.com/bendmorris/taxiphy) - Common repository for scripts to generate trees from taxonomies. Currently includes ITIS, NCBI, and GBIF.
- [gtaxon](https://github.com/shenwei356/gtaxon) - A fast cross-platform NCBI taxonomy data querying (gi2taxid, taxid2taxon, name2taxid, LCA) tool, with cmd client and REST API server for both local and remote server. 
- [R] [taxize](https://github.com/ropensci/taxize) - A taxonomic toolbelt for R http://ropensci.org/tutorials/taxize.html
- [TaxonKit](https://github.com/shenwei356/taxonkit) - Cross-platform and Efficient NCBI Taxonomy Toolkit http://bioinf.shenwei.me/taxonkit/

## Assembly

- [Bandage](https://github.com/rrwick/Bandage) - a Bioinformatics Application for Navigating De novo Assembly Graphs Easily
- [nucleotid.es](http://nucleotid.es) - an assembler catalogue


## Alignment

- [hpg-aligner](https://github.com/opencb/hpg-aligner) - HPG Aligner is an ultrafast and highly sensitive Next-Generation Sequencing (NGS) mapper which supoprts both DNA and RNA alignment
- [AliView](https://github.com/AliView/AliView) - Software for aligning viewing and editing dna/aminoacid sequences, intuitive, fast and lightweight. Download and website: [http://www.ormbunkar.se/aliview](http://www.ormbunkar.se/aliview)

## Multiple Alignment

- [msa: An R Package for Multiple Sequence Alignment](http://www.bioinf.jku.at/software/msa/)


## Mapping

## Bacterial comparative genomics

- [TOOLS FOR BACTERIAL COMPARATIVE GENOMICS](http://holtlab.net/2015/02/25/tools-for-bacterial-comparative-genomics/)

## Metagenomics

- [shotmap](https://github.com/sharpton/shotmap) - A Shotgun Metagenome Annotation Pipeline
- [metagenomeSeq](https://github.com/HCBravoLab/metagenomeSeq) - Statistical analysis for sparse high-throughput sequencing
- [mmgenome](https://github.com/MadsAlbertsen/mmgenome) - Tools for extracting individual genomes from metagneomes
- [harvest](https://github.com/marbl/harvest) - suite of core-genome alignment and visualization tools for quickly analyzing thousands of intraspecific microbial genomes.
- [PhyloSift](https://github.com/gjospin/PhyloSift) - Phylogenetic and taxonomic analysis for genomes and metagenomes
- [MetaQuery: Annotation and quantitative analysis of genes in the human gut microbiome](http://metaquery.docpollard.org/)

- [Microbial Ecology - a discussion and overview of amplicon sequencing and metagenomics](https://angus.readthedocs.io/en/2019/amplicon_and_metagen.html)
- [Quick Insights from Sequencing Data with sourmash](https://angus.readthedocs.io/en/2019/sourmash.html#)
- [Recovering “genomes” from metagenomes](https://angus.readthedocs.io/en/2019/recovering-rep-genomes-from-mgs.html)

network

- [NetCoMi](https://github.com/stefpeschel/NetCoMi) Network Comparison for Microbial Compositional Data

## 16S

-  [高通量数据处理的一些经验和建议](http://blog.sciencenet.cn/blog-871198-677805.html)
-  [How to choose ordination method, such as PCA, CA, PCoA, and NMDS?](https://www.researchgate.net/post/How_to_choose_ordination_method_such_as_PCA_CA_PCoA_and_NMDS)

### Classifier | removing human reads

- [taxonomer.iobio](http://taxonomer.iobio.io/info.html) - Taxonomer is a kmer-based ultrafast metagenomics tool for assigning taxonomy to sequencing reads from both clinical and environmental samples.
- [BMTagger](ftp://ftp.ncbi.nlm.nih.gov/pub/agarwala/bmtagger/) - Best Match Tagger for removing human reads from metagenomics datasets [paper](https://www.researchgate.net/publication/267988741_BMTagger_Best_Match_Tagger_for_removing_human_reads_from_metagenomics_datasets),[sop](http://hmpdacc.org/doc/HumanSequenceRemoval_SOP.pdf)
- [Centrifuge](http://ccb.jhu.edu/software/centrifuge) - Classifier for metagenomic sequences


## Virome

- [viral-ngs](https://github.com/broadinstitute/viral-ngs) - Viral genomics analysis pipelines

## Chip-seq

- [ChIP-seq-analysis](https://github.com/crazyhottommy/ChIP-seq-analysis)

## Plastform

- [Rabix](https://www.rabix.org/) - Portable Bioinformatics Pipelines
- [bioboxes](https://github.com/bioboxes) - Standards for Interchangeable Bioinformatics Containers
- [Anvi’o](http://merenlab.org/projects/anvio/) is an analysis and visualization platform for ‘omics data. [introduction](http://microbe.net/2015/09/04/anvio-a-new-platform-to-work-with-metagenomic-data/?utm_source=twitterfeed&utm_medium=twitter)

## PCR

- [find_differential_primers](https://github.com/widdowquinn/find_differential_primers) - Scripts to aid the design of differential primers for diagnostic PCR.
- [Primer3-py](http://libnano.github.io/primer3-py/) - Primer3-py is a Python-abstracted API for the popular Primer3 library. The intention is to provide a simple and reliable interface for automated oligo analysis and design.

## HPC

- [hpcgo](https://github.com/drio/hpcgo) - Helping submit jobs to HPC cluster.
- [easy_qsub](https://github.com/shenwei356/easy_qsub)  - Easily submitting PBS jobs with script template. Multiple input files supported.

## Transcriptome

- [De novo transcriptome assembly](https://angus.readthedocs.io/en/2019/transcriptome-assembly-nematostella.html)
- [Annotating and evaluating a de novo transcriptome assembly](https://angus.readthedocs.io/en/2019/dammit_annotation.html)
- [Evaluating your transcriptome assembly](https://dibsi-rnaseq.readthedocs.io/en/latest/evaluation.html)

- [Differential Expression and Visualization in R](https://angus.readthedocs.io/en/2019/diff-ex-and-viz.html)
- [RNA-seq Analysis](https://angus.readthedocs.io/en/2019/RNA-seq-Analysis.html)

## variant calling

## one-liner

ensemble id -> symbol -> biotype

    zcat Homo_sapiens.GRCh38.84.gtf.gz \
        | awk '$3=="gene"' \
        | perl -ne 'next unless /gene_id "(.+?)".+gene_name "(.+?)".+gene_biotype "(.+?)"/; print "$1\t$2\t$3\n";' \
        > Homo_sapiens.GRCh38.84.gtf.gz.ensemble2symol-biotype.tsv


## Knowledge

- [Sequencing depth and coverage: key considerations in genomic analyses](https://www.nature.com/articles/nrg3642),
https://www.biostars.org/p/638/, 
[What does coverage mean in the context of NGS?](https://www.ecseq.com/support/ngs/how-to-calculate-the-coverage-for-a-sequencing-experiment)
