# Bioinformatics - Extra Resources

# Software Documentation
* Install CLI tools with [BioConda](https://bioconda.github.io/) package manager
* [Circos](http://circos.ca/documentation/course/)
* [OpenGene Libraries](https://github.com/OpenGene)
* [GATK Tools](https://www.broadinstitute.org/gatk/guide/tooldocs/index)
* [ADAM](http://bdgenomics.org/) Big-Data Genomics
* [FastQC](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/)
* samtools 1:[BioBits'](http://biobits.org/samtools_primer.html) 2:[DaveTang's](https://github.com/davetang/learning_bam_file) 3:[YanhuiFan's](https://felixfan.github.io/bam-sam/) 4:[BinaryFlags](https://broadinstitute.github.io/picard/explain-flags.html)
* [bcftools](https://samtools.github.io/bcftools/howtos/variant-calling.html) 1:[GATK](http://gatkforums.broadinstitute.org/gatk/discussion/1268/what-is-a-vcf-and-how-should-i-interpret-it) 2:[DaveTang's](https://github.com/davetang/learning_vcf_file) 3:[Paper](https://www.researchgate.net/publication/230658044_A_beginners_guide_to_SNP_calling_from_high-Throughput_DNA-sequencing_data)
* bedtools 1:[Quinlan](http://quinlanlab.org/tutorials/bedtools/bedtools.html) 2:[YanhuiFan's](https://felixfan.github.io/bedtools/)
* BWA
* bowtie2
* [Tuxedo Suite](http://cole-trapnell-lab.github.io/cufflinks/) 1:[YanhuiFan's](http://felixfan.github.io/tophat-cufflinks/)
* [bedops](https://bedops.readthedocs.org/en/latest/content/usage-examples.html)
* [picard](http://broadinstitute.github.io/picard/)
* [jvarkit](https://github.com/lindenb/jvarkit)
* [Exomiser](http://www.sanger.ac.uk/science/tools/exomiser) disease causing variants
* [PopSV](https://github.com/jmonlong/PopSV) structural variants
* [Silva](http://compbio.cs.toronto.edu/silva/) silent mutations
* [Variant Effect Predictor](http://www.ensembl.org/info/docs/tools/vep/index.html)
* [ANNOVAR](http://annovar.openbioinformatics.org/en/latest/)
* [BioGPS](http://biogps.org/help/)
* [ABySS](http://sjackman.ca/abyss-activity/)
* [GEMINI](https://gemini.readthedocs.io/en/latest/) annotation db
* Graphical User Interface (GUI) apps
  * [Integrative Genomics Viewer (IGV)](https://www.broadinstitute.org/software/igv/)
  * [NCBI Genome Workbench](https://www.ncbi.nlm.nih.gov/tools/gbench/)
  * [Savant](http://genomesavant.com/p/savant/index)
  * [JBrowse](http://jbrowse.org/)

# Other Non-specific Tools
* [Atom](http://atom.io/) modern text editor
* [Reference Manager](https://en.wikipedia.org/wiki/Comparison_of_reference_management_software)
* [Git](https://git-scm.com/) Version Control
  * [HowTo Guide](https://githowto.com/)
  * [Try Online](https://try.github.io/)
  * [Pro Book](https://progit.org/)
  * [Short Videos](https://git-scm.com/videos)
  * [Start Contributing Guide](https://about.gitlab.com/2016/06/16/fearless-contribution-a-guide-for-first-timers/)
* [LaTeX](https://latex-project.org/intro.html)
* [Overleaf](http://www.overleaf.com/) write collaboratively
* [Markdown](http://daringfireball.net/projects/markdown/)
  * [Here](http://markdown-here.com/)
* [RegEx Generator](http://www.regexr.com/)
* [RegEx 101](https://regex101.com/)
* [RegEx Debugger](https://www.debuggex.com/)
* [Plotly](https://plot.ly/)
* [Small (bioinformatic) Tools Manifesto](https://github.com/pjotrp/bioinformatics)
* [Terminal Multiplexer](https://robots.thoughtbot.com/a-tmux-crash-course)
* [ZSH](http://ohmyz.sh/)
* [VIM](http://vim.rtorr.com/) o rather run 'vimtutor' :P
* [ShellCheck your scripts](http://www.shellcheck.net/)
* Writing
    * [Correct your Grammar](https://www.grammarly.com/)
    * [Look-up word by its meaning](http://www.onelook.com/reverse-dictionary.shtml)
    * [LanguageTool](https://languagetool.org/)
    * google translate, obviusly.
* [Introduction to HTML](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Introduction)
